### PR TITLE
feat(web): Relax GrpcWebService request body type

### PR DESCRIPTION
**Pull request by tottoto**
Relaxes `GrpcWebService`'s request body type.

### CODE REVIEW

File: tonic-web/src/service.rs [hunk 1]
```diff
@@ -113,7 +116,7 @@ where
                 debug!(kind = "other h2", content_type = ?req.headers().get(header::CONTENT_TYPE));
                 ResponseFuture {
                     case: Case::Other {
-                        future: self.inner.call(req),
+                        future: self.inner.call(req.map(tonic::body::boxed)),
```
Comment (tobz) [inline tonic-web/src/service.rs:119]: Hmm, if we have to box the request no matter, what are we gaining with this change?

Comment (tottoto) [reply tonic-web/src/service.rs:119]: The benefit is the `CorsGrpcWeb` can handle any kind of body types which has the trait bounds.

Comment (tobz) [reply tonic-web/src/service.rs:119]: Fair enough. 👍🏻

**Review by tobz — COMMENTED** [review]
While the change itself seems fine, I'm still left with the thought: should we avoid all of the constraints that require a service that accepts `Request<BoxBody>`?

It seems like if we're going to bother to be totally generic over the request body type in `GrpcWebService`, we should also do so in `CorsWebLayer`? Maybe there's a very obvious reason why we can't, but looking at `tower_http::cors::Cors` shows that it's _also_ generic over the request body... so it feels like we should be able to?

**Comment by tottoto** [comment]
> While the change itself seems fine, I'm still left with the thought: should we avoid all of the constraints that require a service that accepts Request<BoxBody>?

> It seems like if we're going to bother to be totally generic over the request body type in GrpcWebService, we should also do so in CorsWebLayer? Maybe there's a very obvious reason why we can't, but looking at tower_http::cors::Cors shows that it's also generic over the request body... so it feels like we should be able to?

Since `CorsWebLayer` does not exist, I respond with information about other parts.

I assume your main question is whether we can avoid requiring a `BoxBody` in the internal services.

I guess so too. When I was working on this, I was also looking into the implementation of `tower_http`'s `Cors` service to see if it is possible to avoid the `BoxBody`. However, it would require an internal design change to the implementation, so I opened this partial change which can be easily done for now.

**Comment by tottoto** [comment]
> While the change itself seems fine, I'm still left with the thought: should we avoid all of the constraints that require a service that accepts Request<BoxBody>?

> It seems like if we're going to bother to be totally generic over the request body type in GrpcWebService, we should also do so in CorsWebLayer? Maybe there's a very obvious reason why we can't, but looking at tower_http::cors::Cors shows that it's also generic over the request body... so it feels like we should be able to?

Since `CorsWebLayer` does not exist, I respond with information about other parts.

I assume your main question is whether we can avoid requiring a `BoxBody` in the internal service.

I guess so too. When I was working on this, I was also looking into the implementation of `tower_http`'s `Cors` service to see if it is possible to avoid the `BoxBody`. However, it would require an internal design change to the implementation, so I opened this partial change which can be easily done for now.

**Comment by tottoto** [comment]
@tobz Are there any other concerns block this? If there are no problems with this, I will merge it at this stage.

**Review by tobz — APPROVED** [review]
Sorry for the delay here.

I do think it's fine for the moment and we can always relax the bounds further in the future.

**Pull request merged**